src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,100 @@
+package duke.storage;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import duke.task.Task;
+import duke.task.ToDo;
+import duke.task.Deadline;
+import duke.task.Event;
+
+/**
+ * Class for reading and writing Duke data.
+ */
+public class Storage {
+    /**
+     * Charset for reading and writing .txt files.
+     */
+    final static Charset ENCODING = StandardCharsets.UTF_8;
+    /**
+     * Path for data.
+     */
+    private Path path;
+
+    /**
+     * Constructor for Storage.
+     *
+     * @param path path for duke.txt
+     * @throws IOException
+     */
+    public Storage(String path) throws IOException {
+        this.path = Paths.get(path);
+    }
+
+    /**
+     * Reads text file from path.
+     *
+     * @return text lines from .txt as List of Strings
+     * @throws IOException
+     */
+    private List<String> read() throws IOException {
+        return Files.readAllLines(path, ENCODING);
+    }
+
+    /**
+     * Writes text file to path.
+     *
+     * @param lines text lines to write onto .txt as List of Strings
+     * @throws IOException
+     */
+    private void write(List<String> lines) throws IOException {
```

Comment:
If the action is one-liner, could it be just written inside writeDuke? I see some clean code pattern here nonetheless.